[bug]: Root page fails to load with status code `404` in Desktop App on Windows

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current behavior

![Image](https://github.com/user-attachments/assets/92745e85-7bca-4a4e-9b09-2074572f912b)

### Steps to reproduce

![Image](https://github.com/user-attachments/assets/b6b2d03e-37c2-4c08-9c72-19efd28d0dc6)

### Environment

Production

### Version

Cloud

I'm also seeing this.

![Image](https://github.com/user-attachments/assets/978b5292-73e6-4a64-b3c0-f37a370dc1b4)

I tried doing the dev mode thing but I am unable to build the app.

<details><summary>Details about the failing build</summary>
<p>

Part of the `pnpm install` process is running `pnpm run postinstall` in `packages\hoppscotch-backend`. This fails with a lot of errors relating to prisma:

```
> prisma generate && pnpm run generate-gql-sdl

Prisma schema loaded from prisma\schema.prisma

✔ Generated Prisma Client (v5.20.0) to .\..\..\node_modules\.pnpm\@prisma+client@5.20.0_prisma@5.20.0\node_modules\@prisma\client in 250ms

Start by importing your Prisma Client (See: http://pris.ly/d/importing-client)

Tip: Need your database queries to be 1000x faster? Accelerate offers you that and more: https://pris.ly/tip-2-accelerate


> hoppscotch-backend@2025.3.1 generate-gql-sdl C:\dev\projects\hoppscotch\packages\hoppscotch-backend
> cross-env GQL_SCHEMA_EMIT_LOCATION='../../../gql-gen/backend-schema.gql' GENERATE_GQL_SCHEMA=true WHITELISTED_ORIGINS='' nest start

src/access-token/access-token.service.ts:13:10 - error TS2305: Module '"@prisma/client"' has no exported member 'PersonalAccessToken'.

13 import { PersonalAccessToken } from '@prisma/client';
            ~~~~~~~~~~~~~~~~~~~
src/auth/auth.service.ts:29:10 - error TS2305: Module '"@prisma/client"' has no exported member 'VerificationToken'.

29 import { VerificationToken } from '@prisma/client';
            ~~~~~~~~~~~~~~~~~
src/health/health.controller.ts:21:59 - error TS2345: Argument of type 'PrismaService' is not assignable to parameter of type 'PrismaClient'.
  Property '$queryRawUnsafe' is missing in type 'PrismaService' but required in type 'PrismaClientSQL'.

21       async () => this.prismaHealth.pingCheck('database', this.prisma),
                                                             ~~~~~~~~~~~

  ../../node_modules/.pnpm/@nestjs+terminus@10.2.3_@ne_5056148dfa9f33df69944cc81ad23f6b/node_modules/@nestjs/terminus/dist/health-indicator/database/prisma.health.d.ts:9:5
    9     $queryRawUnsafe: (query: string) => any;
          ~~~~~~~~~~~~~~~
    '$queryRawUnsafe' is declared here.
src/infra-config/helper.ts:11:10 - error TS2305: Module '"@prisma/client"' has no exported member 'InfraConfig'.

11 import { InfraConfig } from '@prisma/client';
            ~~~~~~~~~~~
src/infra-config/infra-config.service.ts:4:10 - error TS2305: Module '"@prisma/client"' has no exported member 'InfraConfig'.

[...]
```

I googled and fiddled around a bit and decided to copy `packages\hoppscotch-backend\node_modules\.prisma` to `packages\hoppscotch-backend\node_modules\@prisma\client\.prisma` because I saw some paths in `@prisma\client` that seemed to refer to `.prisma`. (I'm not really familiar with nest and prisma.)

But I'm still getting an error:

```
> hoppscotch-backend@2025.3.1 generate-gql-sdl C:\dev\projects\hoppscotch\packages\hoppscotch-backend
> cross-env GQL_SCHEMA_EMIT_LOCATION='../../../gql-gen/backend-schema.gql' GENERATE_GQL_SCHEMA=true WHITELISTED_ORIGINS='' nest start

C:\dev\projects\hoppscotch\packages\hoppscotch-backend\dist\user\user.resolver.js:68
    __metadata("design:paramtypes", [typeof (_a = typeof AuthUser_1.AuthUser !== "undefined" && AuthUser_1.AuthUser) === "function" ? _a : Object]),
                                                         ^

ReferenceError: AuthUser_1 is not defined
    at Object.<anonymous> (C:\dev\projects\hoppscotch\packages\hoppscotch-backend\dist\user\user.resolver.js:68:58)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at Module.load (node:internal/modules/cjs/loader:1206:32)
    at Module._load (node:internal/modules/cjs/loader:1022:12)
    at Module.require (node:internal/modules/cjs/loader:1231:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (C:\dev\projects\hoppscotch\packages\hoppscotch-backend\dist\user\user.module.js:11:25)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)

Node.js v20.12.1
 ELIFECYCLE  Command failed with exit code 1.
 ELIFECYCLE  Command failed with exit code 1.
```
</p>
</details> 

Can you think of another way to shed some light on this?

Is there maybe a way to check which filters have been registered using `AddWebResourceRequestedFilter`?

--- a/packages/hoppscotch-desktop/plugin-workspace/tauri-plugin-appload/Cargo.toml
+++ b/packages/hoppscotch-desktop/plugin-workspace/tauri-plugin-appload/Cargo.toml
@@ -34,6 +34,7 @@ futures = "0.3.31"
 mime_guess = "2.0.5"
 rayon = "1.10.0"
 hex_color = "3.0.0"
+dunce = "1.0.5"
 
 [build-dependencies]
 tauri-plugin = { version = "2.0.1", features = ["build"] }

--- a/packages/hoppscotch-desktop/plugin-workspace/tauri-plugin-appload/src/storage/manager.rs
+++ b/packages/hoppscotch-desktop/plugin-workspace/tauri-plugin-appload/src/storage/manager.rs
@@ -156,12 +156,21 @@ impl StorageManager {
         // NOTE: There cannot be more than one user config storage disk,
         // although even if there is, defaulting to the first one we found
         // is as good of a guess as any.
-        let disk = disks
-            .into_iter()
-            .find(|disk| storage_path.starts_with(disk.mount_point()));
+        let disk = disks.into_iter().find(|disk| {
+            // Convert both paths to the same format for comparison, Windows...
+            let normalized_storage =
+                dunce::canonicalize(&storage_path).unwrap_or(storage_path.clone());
+            let normalized_disk = dunce::canonicalize(disk.mount_point())
+                .unwrap_or_else(|_| disk.mount_point().to_path_buf());
+
+            normalized_storage.starts_with(&normalized_disk)
+        });
 
         let Some(disk) = disk else {
-            tracing::error!("Fatal error, unable to resolve user config storage disk");
+            tracing::error!(
+                storage_path = %storage_path.display(),
+                "Fatal error, unable to resolve user config storage disk"
+            );
             return Err(StorageError::DiskNotFound);
         };
 

--- a/packages/hoppscotch-desktop/plugin-workspace/tauri-plugin-appload/src/ui/windows/posit.rs
+++ b/packages/hoppscotch-desktop/plugin-workspace/tauri-plugin-appload/src/ui/windows/posit.rs
@@ -65,7 +65,7 @@ impl<R: Runtime> WindowsWindow<R> {
 
     fn set_dark_mode(&self) {
         if let Some(version) = WindowsVersion::detect() {
-            if version.major() >= 10 && version.build() >= MIN_WIN11_BUILD {
+            if version.major >= 10 && version.build >= MIN_WIN11_BUILD {
                 unsafe {
                     let use_dark_mode = BOOL::from(true);
                     let _ = DwmSetWindowAttribute(
@@ -81,7 +81,7 @@ impl<R: Runtime> WindowsWindow<R> {
 
     fn set_caption_color(&self, color: HexColor) {
         if let Some(version) = WindowsVersion::detect() {
-            if version.major() >= 10 && version.build() >= MIN_WIN11_BUILD {
+            if version.major >= 10 && version.build >= MIN_WIN11_BUILD {
                 unsafe {
                     let color_ref = self.hex_color_to_colorref(color);
                     let _ = DwmSetWindowAttribute(

